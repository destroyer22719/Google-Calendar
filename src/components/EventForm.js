import React, { useReducer, useState } from "react";
import DateTimePicker from "react-datetime-picker"
import { participantReducer } from "../reducers/participantReducer";
import ParticipantsForm from "./ParticipantsForm";

require('dotenv').config()


const EventForm = (props) => {

	const gapi = window.gapi;
	const CLIENT_ID = process.env.REACT_APP_CLIENT_ID;
	const API_KEY = process.env.REACT_APP_API_KEY;

	const DISCOVERY_DOCS = ["https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest"];
	const SCOPES = "https://www.googleapis.com/auth/calendar.events";

	const [summary, summaryChange] = useState("");
	const [desc, descChange] = useState(""); 
	const [start, onStartChange] = useState(props.start || new Date()); 
	const [end, onEndChange] = useState(start); 
    const [state, dispatch] = useReducer(participantReducer, []);

	var eventExample = {
		'summary': 'Google I/O 2015',
		'location': '800 Howard St., San Francisco, CA 94103',
		'description': 'A chance to hear more about Google\'s developer products.',
		'start': {
		  'dateTime': '2015-05-28T09:00:00-07:00',
		  'timeZone': 'America/Los_Angeles'
		},
		'end': {
		  'dateTime': '2015-05-28T17:00:00-07:00',
		  'timeZone': 'America/Los_Angeles'
		},
		'recurrence': [
		  'RRULE:FREQ=DAILY;COUNT=2'
		],
		'attendees': [
		  {'email': 'lpage@example.com'},
		  {'email': 'sbrin@example.com'}
		],
		'reminders': {
		  'useDefault': false,
		  'overrides': [
			{'method': 'email', 'minutes': 24 * 60},
			{'method': 'popup', 'minutes': 10}
		  ]
		}
	};

	const event = {
		summary: summary,
		description: desc,
		start: {
			dateTime: start.toISOString(),
			timeZone: "America/Toronto"
		},
		end: {
			dateTime: end.toISOString(),
			timeZone: "America/Toronto"
		},
		attendees: state
	}

	const addEvent = settings => {
		gapi.load("client:auth2", async () => {
			gapi.client.init({
				apiKey: API_KEY,
				clientId: CLIENT_ID,
				discoveryDocs: DISCOVERY_DOCS,
				scope: SCOPES,
			});
			gapi.client.load("calendar", "v3", () => console.log("loaded"));
			gapi.auth2.getAuthInstance().signIn()
				.then(() => {
					console.log("Signed in")
					const request = gapi.client.calendar.events.insert({
						"calendarId": "primary",
						"resource": settings,
					});

					request.execute(event => {
						console.log(event.htmlLink)
						window.open(event.htmlLink);
					});
				});
		});
	};

	return (
		<div>
			<h2>Start</h2>
			<DateTimePicker
				onChange={onStartChange}
				value={start}
			/>
			<br/><h2>End</h2>
			<DateTimePicker
				onChange={onEndChange}
				value={end}
			/><br/><label for="summary">Summary</label>
			<input type="text" id="summary" onChange={e => props.summaryChange ? props.summaryChange(e.target.value) : summaryChange(e.target.value)}/>
			<br/><label for="description">Description</label>
			<input type="text" id="description" onChange={e => props.descChange ? props.descChange(e.target.value) : descChange(e.target.value)}/>
			<button id="addEvent" onClick={() => addEvent(event)}>Add Event</button>
			<ParticipantsForm state={state} dispatch={dispatch}/>
		</div>
	);
}

export default EventForm;
