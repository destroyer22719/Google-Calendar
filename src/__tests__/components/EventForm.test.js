import React from "react";
import {shallow} from "enzyme";
import EventForm from "../../components/EventForm";

let wrapper, summaryChange, descChange;

beforeEach(() => {
    summaryChange = jest.fn();
    descChange = jest.fn();

    wrapper = shallow(
        <EventForm 
            summaryChange={summaryChange}
            descChange={descChange}
            start={new Date("Thu Feb 25 2021 10:42:04 GMT-0500 (Eastern Standard Time)")}
        />
    );
});

test("Should render component", () => {
    expect(wrapper).toMatchSnapshot();
});

test("Should handle summary change", () => {
    wrapper.find("#summary").simulate("change", {
        target: {value: "Hello World"}
    });

    expect(summaryChange).toBeCalledWith("Hello World");
});

test("Should handle description change", () => {
    wrapper.find("#description").simulate("change", {
        target: {value: "Hello World! Visit us!"}
    });

    expect(descChange).toBeCalledWith("Hello World! Visit us!");
});

