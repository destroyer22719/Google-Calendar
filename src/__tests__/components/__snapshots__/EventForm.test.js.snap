// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<div>
  <h2>
    Start
  </h2>
  <DateTimePicker
    calendarIcon={
      <svg
        className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
        height={19}
        stroke="black"
        strokeWidth={2}
        viewBox="0 0 19 19"
        width={19}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="15"
          width="15"
          x="2"
          y="2"
        />
        <line
          x1="6"
          x2="6"
          y1="0"
          y2="4"
        />
        <line
          x1="13"
          x2="13"
          y1="0"
          y2="4"
        />
      </svg>
    }
    clearIcon={
      <svg
        className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
        height={19}
        stroke="black"
        strokeWidth={2}
        viewBox="0 0 19 19"
        width={19}
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="15"
          y1="4"
          y2="15"
        />
        <line
          x1="15"
          x2="4"
          y1="4"
          y2="15"
        />
      </svg>
    }
    closeWidgets={true}
    isCalendarOpen={null}
    isClockOpen={null}
    maxDetail="minute"
    onChange={[Function]}
    value={2021-02-25T15:42:04.000Z}
  />
  <br />
  <h2>
    End
  </h2>
  <DateTimePicker
    calendarIcon={
      <svg
        className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
        height={19}
        stroke="black"
        strokeWidth={2}
        viewBox="0 0 19 19"
        width={19}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="15"
          width="15"
          x="2"
          y="2"
        />
        <line
          x1="6"
          x2="6"
          y1="0"
          y2="4"
        />
        <line
          x1="13"
          x2="13"
          y1="0"
          y2="4"
        />
      </svg>
    }
    clearIcon={
      <svg
        className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
        height={19}
        stroke="black"
        strokeWidth={2}
        viewBox="0 0 19 19"
        width={19}
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="15"
          y1="4"
          y2="15"
        />
        <line
          x1="15"
          x2="4"
          y1="4"
          y2="15"
        />
      </svg>
    }
    closeWidgets={true}
    isCalendarOpen={null}
    isClockOpen={null}
    maxDetail="minute"
    onChange={[Function]}
    value={2021-02-25T15:42:04.000Z}
  />
  <br />
  <label
    for="summary"
  >
    Summary
  </label>
  <input
    id="summary"
    onChange={[Function]}
    type="text"
  />
  <br />
  <label
    for="description"
  >
    Description
  </label>
  <input
    id="description"
    onChange={[Function]}
    type="text"
  />
  <button
    id="addEvent"
    onClick={[Function]}
  >
    Add Event
  </button>
  <ParticipantsForm
    dispatch={[Function]}
    state={Array []}
  />
</div>
`;
